twyla/kubedeploy/kube.py

```diff
@@ -16,13 +16,15 @@ def __init__(self,
                  deployment_name: str,
                  printer: Callable[[str], int],
                  error_printer: Callable[[str], int],
-                 deployment_template: str=None):
+                 deployment_template: str=None,
+                 variants: List[str]=[]):
```

Comment:
Using mutables like list or dict in default arguments is one of the most notorious source of subtle bugs in Python. The default idiom for such cases is to have `variants=None` in the function signature, and then set e.g. `self.variants = variants or []`.